Global styles

indirectlylit opened the pull request:
Initial work on shared styles for core and apps:

* global `.styl` sheet
* global `.styl` variables with initial values based on style guide
* embedded NotoSans font
* ability to reference global styles and variables across plugins

This work still needs to be refined a bit to make the shared styling API explicit, but this setup should at least allow us to move forward with development.

MCGallaspy commented on package.json:
```diff
@@ -24,9 +24,10 @@
   "private": true,
   "dependencies": {
-    "vue": "^1.0.21",
+    "babel-runtime": "^6.0.0",
```
Is it time to start pinning dependencies, instead of using `^`?

indirectlylit replied:
Let's all chat about that - I have it as another task. I've been putting it off so we can get the latest versions of libs, and then pin when we're getting closer to release, but I'm open to discussion on timing.

MCGallaspy commented on webpack_config/webpack.config.base.js:
```diff
@@ -14,52 +14,53 @@ var webpack = require('webpack');
+            // shortcut to allow importing the core kolibri_module from other bundles (plugins)
+            'kolibri_module': 'kolibri/core/assets/src/kolibri_module',
+            // Shortcut to allow styles to be imported across all plugins.
+            // Stylus syntax: @require '~theme.styl'
```
You don't use this syntax above, but instead `@require 'theme.styl'`.

MCGallaspy commented on kolibri/plugins/learn/assets/src/learn-main.vue:
```diff
@@ -21,4 +22,12 @@
-<style lang="stylus" scoped></style>
+<style lang="stylus" scoped>
+
+  @require '~theme.styl'
```
So even though `@require 'theme.styl'` appears in core-base, I must still require it here?

MCGallaspy commented:
I have one question -- if as a developer I want to use the global styles, what must I do? You should try to document this somehow.

indirectlylit commented:
Added comments on how to use the global variables.

Also rebased in master, including the new navigation work. As part of that, did some cleanup which we can discuss – see https://github.com/learningequality/kolibri/pull/118/commits/aea300b7aa585878e86dc2d71aae86df82b893b8 (cc @DXCanas)

MCGallaspy commented on kolibri/core/assets/src/navigation.vue:
```diff
@@ -1,16 +1,16 @@
-        <div class="usermenu" v-bind:style="{display: user_nav_display}">
+      <div>
+        <span class="user-menu-btn" v-on:click="userNavDisplayToggle">{{user.first_name}}</span>
+        <div class="user-menu-popup" v-show="userNavDisplay">
```
You'll want to use `nav` tags instead of this `div`, methinks,

MCGallaspy commented on kolibri/core/assets/src/navigation.vue:
```diff
@@ -1,16 +1,16 @@
 
-  <header id="navigation-module">
+  <header>
     <nav class="titlebar">
```
This in fact should not be a `nav` tag, since it doesn't contain navigation items or a list as its immediate children. It seems likely you could remove this element entirely. A good project for @DXCanas in a future PR.
